File: src/dto/diapositiva.dto.ts
import { Colore } from './colore.obj';
import { PresentazioneDTO } from './presentazione.dto';

export class DiapositivaDTO{
  id: number;
  sfondo: Colore;
  ordine: number;
  preset: boolean;
  ratio: string;
  isTitolo: boolean;
  titolo: string;
  posizioneT: string;
  dimensioneT: number;
  coloreT: Colore;
  presentazione: PresentazioneDTO;
  fontFamily: number;
  fontStyle: string;
  fontRotation: number;
  borderSize: number;
  borderColor: Colore;

  constructor(id: number, sfondo: Colore, ordine: number, preset: boolean, ratio: string,
              isTitolo: boolean, titolo: string, posizioneT: string,
              dimensioneT: number, coloreT: Colore, presentazione: PresentazioneDTO,
              fontFamily?: number, fontStyle?: string, fontRotation?: number, borderSize?: number, borderColor?: Colore ){
    this.id = id;
    this.sfondo = sfondo;
    this.ordine = ordine;
    this.preset = preset;
    this.ratio = ratio;
    this.isTitolo = isTitolo;
    this.titolo = titolo;
    this.posizioneT = posizioneT;
    this.dimensioneT = dimensioneT;
    this.coloreT = coloreT;
    this.presentazione = presentazione;
    this.fontFamily = fontFamily ? fontFamily : 0;
    this.fontStyle = fontStyle ? fontStyle : 'normal';
    this.fontRotation = fontRotation ? fontRotation : 0;
    this.borderSize = borderSize ? borderSize : 0;
    this.borderColor = borderColor ? borderColor : new Colore(255,255,255,100);

  }
}

File: Presentazione/Presentazione/DTO/DiapositivaDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using Presentazione.Models;

namespace Presentazione.DTO
{
    public class DiapositivaDTO
    {
        public long id { get; set; }

        public PresentazioneDTO presentazione { get; set; }

        public Colore sfondo { get; set; }
        
        public int ordine { get; set; }

        public Boolean preset { get; set; }

        public string ratio { get; set; }

        public Boolean isTitolo { get; set; }

        public string titolo { get; set; }

        public string posizioneT { get; set; }

        public int dimensioneT { get; set; }

        public Colore coloreT { get; set; }

        public int fontFamily { get; set; }

        public string fontStyle { get; set; }

        public int fontRotation { get; set; }

        public int borderSize { get; set; }

        public Colore borderColor { get; set; }

    }
}
